File: CompositeVisitor/CompositeVisitor_java/ShapePrinterVisitor.java
public class ShapePrinterVisitor implements IShapeVisitor {
  private int m_tabCount;
  
  public ShapePrinterVisitor() {
    m_tabCount = 0;
  }
  
  private void printTabs() {
    for (int i = 0; i < m_tabCount; i++) {
      System.out.print("\t");
    }
  }
  
  public void visitRectangle(Rectangle a_rect) {
    printTabs();
    System.out.println("Rectanlge Area: " + a_rect.getArea());
  }
  
  public void visitCircle(Circle a_circle) {
    printTabs();
    System.out.println("Circle Area: " + a_circle.getArea());
  }
  
  public void visitCompositePre(CompositeShape a_group) {
    printTabs();
    System.out.println("Composite Total Area: " + a_group.getArea());
    System.out.println("Composite parts: ");
    m_tabCount++;
  }
  
  public void visitCompositePost(CompositeShape a_group) {
    m_tabCount--;
  }
    
}

File: CompositeVisitor/CompositeVisitor_Swift_3.0.1/CompositeVisitor/view/ShapePrinterVisitor.swift
//
//  ShapePrinterVisitor.swift
//  CompositeVisitor
//
//  Created by Sunny Buttar on 2016-11-13.
//  Copyright © 2016 Sarpreet Singh. All rights reserved.
//

import Foundation

class ShapePrinterVisitor: IShapeVisitor {
    
    private var m_tabCount : Int;
    
    init() {
        m_tabCount = 0;
    }
    
     func printTabs() {
        for _ in 0...m_tabCount {
            print("\t");
        }
    }
    
    func visitRectangle(a_rect: Rectangle) {
        printTabs();
        print("Rectanlge Area: " + String (a_rect.getArea()));
    }
    
    func visitCircle(a_circle: Circle) {
        printTabs();
        print("Circle Area: " + String (a_circle.getArea()));
    }
    
    func visitCompositePre(a_group: CompositeShape) {
        printTabs();
        print("Composite Total Area: " + String (a_group.getArea()));
        print("Composite parts: ");
        m_tabCount += 1;
    }
    
    func visitCompositePost(a_group: CompositeShape) {
        m_tabCount -= 1;
    }
}
